ci/__main__.py

```diff
@@ -0,0 +1,686 @@
+# dependencies. Packages are forced re-installed in this order.
+source_dependencies = {
+    "pyface": "git+http://github.com/enthought/pyface.git#egg=pyface",
+    "traits": "git+http://github.com/enthought/traits.git#egg=traits",
+}
+
+# # The following should match extras_require in setup.py but with package
+# # names compatible with EDM
+# extra_dependencies = {
+#     'pyside2': {
+#         'pygments',
+#     },
+#     'pyside6': {
+#         'pygments',
+#     },
+#     'pyqt5': {
+#         'pyqt5',
+#         'pygments',
+#     },
+#     'pyqt6': {
+#         'pygments',
+#     },
+#     # XXX once wxPython 4 is available in EDM, we will want it here
+#     'wx': {
+#         'numpy',
+#     },
+#     'null': set(),
+#     # Toolkit-independent dependencies for demo testing.
+#     'examples': {
+#         'apptools',
+#         'chaco',
+#         'h5py',
+#         'numpy',
+#         'pandas',
+#         'pytables',
+#     },
+#     # Optional dependencies for some editors
+#     'editors': {
+#         'numpy',
+#         'pandas',
+#         'pillow',
+#     },
+#     # Test dependencies also applied to installation from PYPI
+#     'test': {
+#         'packaging',
+#     },
+# }
+
+# Dependencies for CI jobs using this file.
+ci_dependencies = {
+    "coverage",
+    "flake8",
+    "flake8_ets",
+}
+
+# doc_dependencies = {
+#     "sphinx",
+#     "enthought_sphinx_theme",
+#     "pyface",
+# }
+
+#: Paths to ignore in Sphinx-apidoc
+# doc_ignore = {
+#     "traitsui/qt4/*",
+#     "traitsui/wx/*",
+#     "*/tests",
+# }
+
+environment_vars = {
+    # 'pyside2': {'ETS_TOOLKIT': 'qt', 'QT_API': 'pyside2'},
+    # 'pyside6': {'ETS_TOOLKIT': 'qt', 'QT_API': 'pyside6'},
+    'pyqt5': {"ETS_TOOLKIT": "qt", "QT_API": "pyqt5"},
+    # 'pyqt6': {"ETS_TOOLKIT": "qt", "QT_API": "pyqt6"},
+    # 'wx': {'ETS_TOOLKIT': 'wx'},
+    # 'null': {'ETS_TOOLKIT': 'null'},
+}
+
+# Location of documentation files
+HERE = os.path.dirname(__file__)
+DOCS_DIR = os.path.join(HERE, "docs")
+
+# Location of news fragment for creating changelog.
+NEWS_FRAGMENT_DIR = os.path.join(DOCS_DIR, "releases", "upcoming")
+
+# Location of the Changelog file.
+CHANGELOG_PATH = os.path.join(HERE, "CHANGES.txt")
+
+
+def normalize(name):
+    return name.replace("_", "-")
+
+
+# Ensure that "-h" is supported for getting help.
+CONTEXT_SETTINGS = dict(
+    help_option_names=["-h", "--help"],
+    token_normalize_func=normalize,
+)
+
+
+@click.group(context_settings=CONTEXT_SETTINGS)
+def cli():
+    pass
+
+
+@cli.command()
+@click.option('--runtime', default=DEFAULT_RUNTIME)
+@click.option('--toolkit', default=DEFAULT_TOOLKIT)
+@click.option('--environment', default=None)
+@click.option(
+    "--editable/--not-editable",
+    default=False,
+    help="Install main package in 'editable' mode?  [default: --not-editable]",
+)
+@click.option('--source/--no-source', default=False)
+def install(runtime, toolkit, environment, editable, source):
```

Comment:
When run as is, this command will install the `ets_tutorial` package, and not any of the application ones. It'll be useful to document this fact. And perhaps "setup" or "build" would be a better name for this command.

Reply:
Great idea. Will change to "setup".

Reply:
Actually going to change it to "build", to avoid any confusion with `setuptools` "setup".